tests/integration/components/frost-bunsen-form/renderers/date-test.js

```diff
@@ -0,0 +1,461 @@
+              name: "date"
+            }
+          }
+        ],
+        type: 'form',
+        version: '2.0'
+      })
+    })
+
+    it('renders as expected', function () {
+      expectCollapsibleHandles(0)
+      expectBunsenDateRendererWithState('foo', {label: 'Foo'})
+      expectOnValidationState(ctx, {count: 1})
+    })
+  })
+
+  describe('when form explicitly enabled', function () {
+    beforeEach(function () {
+      this.set('disabled', false)
+    })
+
+    it('renders as expected', function () {
+      expectCollapsibleHandles(0)
+      expectBunsenDateRendererWithState('foo', {label: 'Foo'})
+      expectOnValidationState(ctx, {count: 1})
+    })
+  })
+
+  describe('when form disabled', function () {
+    beforeEach(function () {
+      this.set('disabled', true)
+    })
+
+    it('renders as expected', function () {
+      expectCollapsibleHandles(0)
+      expectBunsenDateRendererWithState('foo', {
+        disabled: true,
+        label: 'Foo'
+      })
+      expectOnValidationState(ctx, {count: 1})
+    })
+  })
+
+  describe('when property explicitly enabled in view', function () {
+    beforeEach(function () {
+      this.set('bunsenView', {
+        cells: [
+          {
+            disabled: false,
+            model: 'foo',
+            renderer: {
+              name: "date"
+            }
+          }
+        ],
+        type: 'form',
+        version: '2.0'
+      })
+    })
+
+    it('renders as expected', function () {
+      expectCollapsibleHandles(0)
+      expectBunsenDateRendererWithState('foo', {label: 'Foo'})
+      expectOnValidationState(ctx, {count: 1})
+    })
+  })
+
+  describe('when property disabled in view', function () {
+    beforeEach(function () {
+      this.set('bunsenView', {
+        cells: [
+          {
+            disabled: true,
+            model: 'foo',
+            renderer: {
+              name: "date"
+            }
+          }
+        ],
+        type: 'form',
+        version: '2.0'
+      })
+    })
+
+    it('renders as expected', function () {
+      expectCollapsibleHandles(0)
+      expectBunsenDateRendererWithState('foo', {
+        disabled: true,
+        label: 'Foo'
+      })
+      expectOnValidationState(ctx, {count: 1})
+    })
+  })
+
+  describe('when date is set', function () {
+    beforeEach(function () {
+      ctx.props.onValidation.reset()
+      const interactor = openDatepickerBunsenDateRenderer('foo')
+      interactor.selectDate(new Date(2017, 0, 24))
+      closePikaday(this)
+    })
+
+    it('functions as expected', function () {
+      expectCollapsibleHandles(0)
+      expectBunsenDateRendererWithState('foo', {
+        hasValue: true,
+        label: 'Foo'
+      })
+      expectOnChangeState(ctx, {foo: '2017-01-24'})
+      expectOnValidationState(ctx, {count: 1})
+    })
+
+    describe('when date is unset', function () {
+      beforeEach(function () {
+        ctx.props.onValidation.reset()
+        this.set('value', {})
```

Comment:
As far as I can tell, it's currently impossible for the user to clear the value of an `ember-frost-date-picker`, so clearing it programmatically will have to do?

https://github.com/ciena-frost/ember-frost-date-picker/issues/26